### PR TITLE
jing-trang: 20151127 -> 20181222

### PR DESCRIPTION
**Pull request by LeSuisse**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>jing</li>
    <li>xmloscopy</li>
  </ul>
</details>

### CODE REVIEW

**Comment by bobby285271** [comment]
@ofborg eval

File: pkgs/tools/text/xml/jing-trang/default.nix [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
+  version = "20181222";
 
   src = fetchFromGitHub {
     owner = "relaxng";
     repo = "jing-trang";
-    rev = "47a0cbdaec2d48824b78a1c19879ac7875509598"; # needed to compile with jdk8
-    sha256 = "1hhn52z9mv1x9nyvyqnmzg5yrs2lzm9xac7i15izppv02wp32qha";
+    rev = "V${version}";
+    sha256 = "sha256-cQ1yd8k8qv8OQxDus1nUU2LUQBswQRXX/Rif88Xl5gY=";
+    leaveDotGit = true;
```
Comment (bjornfor) [inline pkgs/tools/text/xml/jing-trang/default.nix:12]: Is leaveDotGit be reproducible? (I think not.)

Comment (LeSuisse) [reply pkgs/tools/text/xml/jing-trang/default.nix]: I patched the build process so it does not rely on `git`.  `leaveDotGit = true;` is no more needed.

**Review by bjornfor — APPROVED** [review]

**Comment by bjornfor** [comment]
Thanks!

**Pull request merged**